core/src/agent/chain_store.rs

```diff
@@ -38,6 +39,22 @@ where
                 .find(|chain_header| chain_header.entry_type() == entry_type),
         )
     }
+
+    pub fn query(
+        &self,
+        start_chain_header: &Option<ChainHeader>,
```

Comment:
it seems weird that this is in the query here but not exposed in `QueryArgs`

extra weird that `limit` is exposed in `QueryArgs`

so we almost have pagination but not quite.

it makes sense to me that `limit` and `start` would almost always be used together to page through results?

Reply:
'start' and 'end' would make sense yes, but 'limit' is not 'end'.
'end' would be a chainHeader address where you would have to stop iterating, which is different from the limit of number of results the query/find gives back.